build.ps1

```diff
@@ -104,5 +104,5 @@ if (!(Test-Path $nuget))
 & $nuget restore $RepoRoot\sdk-templates.sln
 if($LASTEXITCODE -ne 0) { throw "Failed to restore nuget packages for templates" }
 
-msbuild $commonBuildArgs /nr:false /p:BuildTemplates=true /flp1:Summary`;Verbosity=diagnostic`;Encoding=UTF-8`;LogFile=$msbuildSummaryLog /flp2:WarningsOnly`;Verbosity=diagnostic`;Encoding=UTF-8`;LogFile=$msbuildWarningLog /flp3:ErrorsOnly`;Verbosity=diagnostic`;Encoding=UTF-8`;LogFile=$msbuildFailureLog
+dotnet msbuild $commonBuildArgs /p:BuildTemplates=true /flp1:Summary`;Verbosity=diagnostic`;Encoding=UTF-8`;LogFile=$msbuildSummaryLog /flp2:WarningsOnly`;Verbosity=diagnostic`;Encoding=UTF-8`;LogFile=$msbuildWarningLog /flp3:ErrorsOnly`;Verbosity=diagnostic`;Encoding=UTF-8`;LogFile=$msbuildFailureLog
```

Comment:
@srivatsn Do you see an issue with changing the template build to use core msbuild? I assume this is OK, but I don't know if there's anything special about the templates build that would require desktop msbuild. It's changing here so that we can add a test project modeled after the others that use the SDK and package refs.